filter/src/main/java/no/sb1/troxy/filter/AutoRegexNewRecording.java

```diff
@@ -141,6 +173,57 @@ protected void filterNewRecording(Recording recording) {
         responseTemplate.setContent(getReplacedField(responseTemplate.getContent(), responseContentPatterns, false));
     }
 
+    private String getReplacedFieldLineByLine(String input, List<PatternWithReplacement> patterns) {
+        // do nothing if no pattern
+        if (patterns.isEmpty()) {
+            return input;
+        }
+
+        // unescape input and chop off leading "^" and trailing "$"
+        String text = unescape(input.substring(1, input.length() - 1));
+        if (text == null) {
+            log.warn("Unable to unescape input '{}', can't search & replace text with regular expressions", input);
+            return input;
+        }
+
+        Set<Integer> linesToNotEscape = new HashSet<>();
+        String[] lines = text.split("\n");
+        for (PatternWithReplacement pattern : patterns) {
+            for (int currentLine = 0; currentLine < lines.length; currentLine++) {
```

Comment:
Maybe a tiny comment would be helpful here. Mention that we're going through every header for each regex to see if there's a match.